lib/resources/etc_libuser.rb

```diff
@@ -0,0 +1,108 @@
+require 'utils/simpleconfig'
+require 'utils/object_traversal'
+
+module Inspec::Resources
+  class EtcLibUser < Inspec.resource(1)
+    name 'etc_libuser'
+    desc 'Use the etc_libuser InSpec audit resource to check the configuration of /etc/libuser.conf.
+          Due to limitations of the RSpec engine and for consistency, please use the method described here
+          (https://github.com/chef/inspec/issues/1543) for accessing the hierarchical values in
+          /etc/libuser.conf'
+    example "
+          ---start sample /etc/libuser.conf---
+          # See libuser.conf(5) for more information.
+          ...
+          [import]
+          # Data from these files is used when libuser.conf does not define a value.
+          # The mapping is documented in the man page.
+          login_defs = /etc/login.defs
+          default_useradd = /etc/default/useradd
+          ...
+          ---end sample /etc/libuser.conf---
+
+          describe etc_libuser do
+            its(['import','login_defs']){should eq ['/etc/login.defs']}
+          end
+         "
+
+    include ObjectTraverser
+    attr_reader :params, :raw_content
+
+    class UnknownEtcLibUserConf < StandardError; end
+
+    def initialize(conf_path = nil)
+      @conf_path = conf_path || inspec.os_env('LIBUSER_CONF').content || '/etc/libuser.conf'
+      file = inspec.file(@conf_path)
+      return skip_resource "The `etc_libuser` resource is not supported on your OS. Can't find '#{@conf_path}'." unless file.file?
+      @params = nil
+      @raw_content = load_raw_content(@conf_path)
+      @params = parse(@raw_content)
+    end
+
+    def parse(content)
+      @params = SimpleConfig.new(content, multiple_values: true).params
+    end
+
+    def to_s
+      'Libuser Configuration'
```

Comment:
Should we add the file name here in case a user test multiple libuser configs in the same profile and therefore would need to know which is which?

perhaps:

```ruby
def to_s
  "LibUser Config from #{@conf_path}"
end
```

Reply:
Yeah, I think this is a good idea. I will update the code now.